### PR TITLE
hw3 + hw4

### PR DESCRIPTION
**Pull request by danikkrapivnitskiy**

### CODE REVIEW

File: hw_3/task_2.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+let age = 10;
+const age_2 = 18;
+const age_3 = 60;
+
```
Comment (VladimirLQA) [inline hw_3/task_2.js]: Follow task description - Преобразовать Task 2 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежит ТОЛЬКО ЦИФРА) пропускалось,
  преобразовываясь в number

File: hw_3/task_3.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Age prompt</title>
+</head>
+<body>
+    <script>
+        const minAge = 18;
+        const maxAge = 60;
+
+        let age = prompt('Print the age');
+        age = parseInt(age);
```
Comment (VladimirLQA) [inline hw_3/task_3.html]: Have a look at the comment above

File: hw_4/task_1.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+for (let i = 0; i <= 10; i++) {
+    console.log(i);
+}
```
Comment (VladimirLQA) [inline hw_4/task_1.js]: Pay attention to task description - `output from 10 to 0`

File: hw_4/task_2.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+4. Написать скрипт, который преобразует любое предложение в camelCase. Первое слово должно начинаться с буквы в нижнем регистре, 
+  у остальных -  верхнем. Пример: I am super engineer => iAmSuperEngineer
+*/
+
+// 1.
+const nameList = ["bytes", "Kb", "Mb", "Gb", "Tb"]
+function convertion(bytes) {
+    for (let i = 0; i < nameList.length; i++) {
+        if (bytes < 1000 || (i + 1) == nameList.length) {
+            console.log(bytes.toFixed(2) + " " + nameList[i]);
```
Comment (VladimirLQA) [inline hw_4/task_2.js]: Rounded to 1 digit

File: hw_4/task_3.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+function word(wordSet) {
+    let word = wordSet.toLocaleLowerCase()
+vowels = "aeiou"
+vowels_count = 0
+consonants_count = 0
+for(i of word){
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: Explicit variable declaration

File: hw_4/task_3.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    if(vowels.includes(i)){
+        vowels_count++;
+        }
+   else {
+    consonants_count++;
+        }
+}
+console.log(`word contains vowels: ${vowels_count}, and  consonants: ${consonants_count}`);
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: Try to test this example `ab21`

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: hw_4/task_3.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		consonants_count++;
+	}
+}
+console.log(`word contains vowels: ${vowels_count}, and  consonants: ${consonants_count}`);
+}
+
+word("ab21")
+
+//2
+const str = 'aAbZzyZ';    // 'bBcAazA' for +1 and 'zZaYyxY' for -1
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: Try to modify this solution to handle such case `ab 1Z` - leave space and number in the result line

File: hw_4/task_3.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+function word(wordSet) {
+    let word = wordSet.toLocaleLowerCase()
+vowels = "aeiou"
+vowels_count = 0
+consonants_count = 0
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: Explicit variables declaration, please

File: hw_4/task_3.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+for(const i of word){
+    if(vowels.includes(i)) vowels_count++; 
+	else if (typeof i === "symbol") {
+		consonants_count++;
+	}
+}
+console.log(`word contains vowels: ${vowels_count}, and  consonants: ${consonants_count}`);
+}
+
+word("ab21")
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: With this solution I got such result - `word contains vowels: 1, and  consonants: 0`
but desired result is `word contains vowels: 1, and  consonants: 1`

File: hw_4/task_3.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+function word(wordSet) {
+    let word = wordSet.toLocaleLowerCase()
+vowels = "aeiou"
+vowels_count = 0
+consonants_count = 0
+for(const i of word){
+    if(vowels.includes(i)) vowels_count++; 
+	else if (typeof i === "symbol") {
```
Comment (VladimirLQA) [inline hw_4/task_3.js]: I can't understand where we got or will get value of symbol type

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: hw_3/task_3.html (hunk 2 above)
Comment (VladimirLQA) [reply hw_3/task_3.html]: Use `isNaN` check with further conversion of the promt value for switch cases as in the previos task

File: JSTutorial/hw_4/task_1.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+for (let i = 10; i <= 0; i++) {
+    console.log(i);
+}
```
Comment (VladimirLQA) [inline JSTutorial/hw_4/task_1.js]: Check the `step` part in this loop, had nothing in console, after runnig code

File: JSTutorial/hw_4/task_2.js [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+4. Написать скрипт, который преобразует любое предложение в camelCase. Первое слово должно начинаться с буквы в нижнем регистре, 
+  у остальных -  верхнем. Пример: I am super engineer => iAmSuperEngineer
+*/
+
+// 1.
+const nameList = ["bytes", "Kb", "Mb", "Gb", "Tb"]
+function convertion(bytes) {
+    for (let i = 0; i < nameList.length; i++) {
+        if (bytes < 1000 || (i + 1) == nameList.length) {
+            console.log(bytes.toFixed(2) + " " + nameList[i]);
```
Comment (VladimirLQA) [inline JSTutorial/hw_4/task_2.js]: Have to round the result to 1 number

File: JSTutorial/hw_4/task_2.js [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+    }
+}
+
+rhomb(4)
+
+//3.
+for (let i = 0; i <= 100; i++) {
+    switch (true) {
+        case i % 15 == 0: {
+            console.log("Число делится и на 3 и на 5");
```
Comment (VladimirLQA) [inline JSTutorial/hw_4/task_2.js]: Add the value to console log that meet condition `15 - Число делится и на 3 и на 5`
Same for other cases

File: JSTutorial/hw_4/task_3.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
+	function word(wordSet) {
+		let word = wordSet.toLocaleLowerCase()
```
Comment (VladimirLQA) [inline JSTutorial/hw_4/task_3.js]: Why not use `toLowerCase()` ?

File: JSTutorial/hw_5/task_1.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+2. В функции создайте массив имен сотрудников 5шт (Имена придумать самим)
+3. В функции создайте массив зарплат сотрудников 5 шт(цифры придумать самим)
+4. Функция должна принимать 1 аргумент - имя сотрудника
+5. Функция должна возвращать новый массив, где первый элемент - имя сотрудника, второй - его зарплата
+6. Для поиска ответа функции нужно найти индекс сотрудника в массиве имен. Зарплату взять с ТЕМ ЖЕ индексом что и имя
+7. Для возврата из функции создайте массив, методом .push поместите в него имя и зарплату, и через return верните созданный массив
+8. Если такое имя сотрудника в массиве не найдется - вернуть null
+ */
+
+function getEmployeeInfo(name) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_1.js:12]: Add validation that passed argument should be a string type, if not throw an error

File: JSTutorial/hw_5/task_2.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+4. Написать функцию, которая принимает на вход слово и проверяет, является ли это слово палиндромом
+
+ */
+
+// 1.
+const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai']
+const myPizza = ['Peperoni', 'hawai', 'machette', 'caprioni', 'diablo']
+
+function checkNotExistCompetitor(pizzaArray) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js:20]: Add validation that passed argument should be an array, if not throw an error
Same for other functions ....

File: JSTutorial/hw_5/task_2.js [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+    const array = []
+
+    for (let i of pizzaArray) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
    for (const i of pizzaArray) {
```

File: JSTutorial/hw_5/task_2.js [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+
+console.log(checkNotExistCompetitor(myPizza))
+
+
+//2.
+
+function getLargestWord(sentense) {
+    const array = sentense.split(" ")
+    let largestWord = []
+    let maxCount = 0;
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
    let wordLength = 0;
```

File: JSTutorial/hw_5/task_2.js [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+
+    for (let word of array) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
    for (const word of array) {
```

File: JSTutorial/hw_5/task_2.js [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+        if (word.length > maxCount) {
+            largestWord = []
+            largestWord.push(word);
+            maxCount = word.length;
+        } else if (word.length == maxCount) largestWord.push(word);
+    }
+    
+    return largestWord.length > 0 ? largestWord : null
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
        if (word.length > wordLength) {
            largestWord = [word]
            wordLength = word.length;
        } else if (word.length == wordLength) { 
             largestWord.push(word);
         }
    }
    
    return largestWord.length > 0 ? largestWord : null
```

File: JSTutorial/hw_5/task_2.js [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+console.log(getLargestWord("Big VeryBig 2timesbigger 2timesbigger"))
+
+//3.
+const arr = [1, 3, 5, 3, 1, 6, 10]
+
+function removeDublicates(array) {
+    const newArray = []
+
+    for (let symbol of array) {
+        if (!newArray.includes(symbol) && !isNaN(symbol)) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
        if (!newArray.includes(symbol)) {
```
Just remove elements that appear two and more times

File: JSTutorial/hw_5/task_2.js [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+    const arr = [...word]
+    const reversedArray = arr.slice().reverse();
+    let isEqual = true;
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] !== reversedArray[i]) {
+        isEqual = false;
+        break;
+        }
+    }
+    return isEqual;
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_2.js]:
```suggestion
    const reversedArray = arr.slice().reverse();
    for (let i = 0; i < arr.length / 2; i++) {
        if (arr[i] !== reversedArray[i]) {
           return false;
        }
    }
    return true;
```

Divide the arr.length to make less iterations simplify returns

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: JSTutorial/hw_4/task_2.js (hunk 12 above)
Comment (danikkrapivnitskiy) [reply JSTutorial/hw_4/task_2.js]: why? result rounding 98798769786 is 98.80 -> correct

Comment (VladimirLQA) [reply JSTutorial/hw_4/task_2.js]: from task description
Результат должен быть округлен до 1 знака после запятой методом .toFixed(), про который надо почитать самим ;)

File: JSTutorial/hw_5/task_1.js [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+    if (!isNaN(name)) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_5/task_1.js]:
```suggestion
    if (typeof name === 'string') {
```

File: JSTutorial/hw_4/task_3.js [hunk 24]
```diff
@@ -0,0 +1,48 @@
+	function word(wordSet) {
+		let word = wordSet.toLowerCase()
+		const vowels = "aeiou"
+		let vowels_count = 0
+		let consonants_count = 0
+	for(const i of word) {
+		if(vowels.includes(i)) {
+			vowels_count++; 
+		} else if (isNaN(i)) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_4/task_3.js]: Add the string that contains consonants and do `.includes(i)` check
`isNaN` would not work with special characters as `#, $` etc.

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**